scipy/spatial/distance.py

```diff
@@ -1364,6 +1364,11 @@ def pdist(X, metric='euclidean', p=None, w=None, V=None, VI=None):
     # between all pairs of vectors in X using the distance metric 'abc' but
     # with a more succinct, verifiable, but less efficient implementation.
 
+
+    if scipy.sparse.issparse(X):
+         raise NotImplementedError("pdist does not support sparse matrices, use skearn's pairwise_distances\nhttp://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.pairwise_distances.html\n")
```

Comment:
The [docs](https://docs.python.org/3/library/exceptions.html#NotImplementedError) indicate that `NotImplementedError` should be used for abstract methods or in-progress methods. I don't think there's a plan to support sparse inputs here, so `TypeError` might be more appropriate.

Also, I'd like a shorter message. Maybe something like: "pdist doesn't support sparse matrix inputs, use sklearn.metrics.pairwise_distances instead."